build.go

```
@@ -296,25 +298,46 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 	if err != nil {
 		return err
 	}
-
+	// TODO make a method for project descriptor and check how we call here and then do the same for git
 	version := opts.ProjectDescriptor.Project.Version
 	sourceURL := opts.ProjectDescriptor.Project.SourceURL
 	runImageName, err = pname.TranslateRegistry(runImageName, c.registryMirrors, c.logger)
 	if err != nil {
 		return err
 	}
+	var (
+		MetadataObj platform.ProjectSource
+	)
 
-	lifecycleOpts := build.LifecycleOptions{
-		AppPath:        appPath,
-		Image:          imageRef,
-		Builder:        ephemeralBuilder,
-		LifecycleImage: ephemeralBuilder.Name(),
-		RunImage:       runImageName,
-		ProjectMetadata: platform.ProjectMetadata{Source: &platform.ProjectSource{
+	r, error := git.PlainOpen(appPath)
+
+	if error != nil {
+		MetadataObj = platform.ProjectSource{
 			Type:     "project",
 			Version:  map[string]interface{}{"declared": version},
-			Metadata: map[string]interface{}{"url": sourceURL},
-		}},
+			Metadata: map[string]interface{}{"url": sourceURL}}
+	} else {
+		hashCommit, _ := r.Head()
+		tagObj, _ := r.TagObject(hashCommit.Hash())
```

Ignoring errors is not acceptable in this scenario.